### PR TITLE
Correct output for wwctl node list -a for slices

### PR DESCRIPTION
**Pull request by mslacken**
This is freaking change, as normaly I wouldn't touch Entry,
but as Entry will go away in the future, get this done with
this ugly workarround.

- add various tests for `wwctl node list -a`
- Entry is aware if its ment as a slice or not

### CODE REVIEW

**Comment by anderbubble** [comment]
@mslacken I see you've added this to v4.5.0. This needs to go in, delaying the release? (Not an objection; just confirming.)

**Comment by mslacken** [comment]
Just wanted to hear your thoughts about this, delay or release?

**Comment by anderbubble** [comment]
I'm not clear of the user-visible nature of this change. Can you provide an example?

**Comment by mslacken** [comment]
See following output
```
wwctl node list -a dell02
  NODE    FIELD                        PROFILE     VALUE                                               
  dell02  Id                           --          dell02                                              
  dell02  Comment                      SUPERSEDED  superseded comment,Special profile for slurm tests  
  dell02  ContainerName                slurm15sp4  slurm15sp4                                          
  dell02  Ipxe                         --          (default)                                           
  dell02  RuntimeOverlay               --          (generic)
```
So the list of the comment is wrong, although e.g. a template would have gotten the right values.

**Comment by anderbubble** [comment]
@mslacken can you rebase on main?

**Comment by anderbubble** [comment]
@mslacken I still don't understand what this change is doing.

Here's a run of `node list -a` from main:

```
[janderson@rocky main]$ sudo wwctl node list -a n1
  NODE  FIELD                        PROFILE     VALUE                                                                                                                                                                    
  n1    Id                           --          n1                                                                                                                                                                       
  n1    Comment                      SUPERSEDED  a node comment,This profile is automatically included for each node                                                                                                      
  n1    ClusterName                  --          mycluster                                                                                                                                                                
  n1    ContainerName                default     rockylinux-8                                                                                                                                                             
  n1    Ipxe                         --          (default)                                                                                                                                                                
  n1    RuntimeOverlay               --          (generic)                                                                                                                                                                
  n1    SystemOverlay                --          wwinit,fstab                                                                                                                                                             
  n1    Root                         --          (initramfs)                                                                                                                                                              
  n1    Init                         --          (/sbin/init)                                                                                                                                                             
  n1    Kernel.Args                  SUPERSEDED  crashkernel=no vga=791 net.naming-scheme=v238,crashkernel=no vga=791 net.naming-scheme=v238 systemd.unified_cgroup_hierarchy=0 systemd.legacy_systemd_cgroup_controller  
  n1    Profiles                     --          default                                                                                                                                                                  
  n1    PrimaryNetDev                --          (default)                                                                                                                                                                
  n1    NetDevs[default].Type        --          Ethernet                                                                                                                                                                 
  n1    NetDevs[default].OnBoot      --          true                                                                                                                                                                     
  n1    NetDevs[default].Device      --          eth0                                                                                                                                                                     
  n1    NetDevs[default].Hwaddr      --          de:ad:be:ef:00:00                                                                                                                                                        
  n1    NetDevs[default].Ipaddr      --          192.168.0.100                                                                                                                                                            
  n1    NetDevs[default].Netmask     --          255.255.255.0                                                                                                                                                            
  n1    NetDevs[default].Gateway     --          192.168.0.1                                                                                                                                                              
  n1    NetDevs[default].Primary     --          (true)                                                                                                                                                                   
  n1    NetDevs[default].Tags[DNS1]  SUPERSEDED  8.8.8.8,8.8.8.8                                                                                                                                                          
  n1    NetDevs[default].Tags[DNS2]  SUPERSEDED  8.8.4.4,8.8.4.4                                                                                                                                                          
  n1    NetDevs[default].Tags[DNS]   --          8.8.8.8                                                                                                                                                                  
  n1    Tags[slurmdconf]             default     Feature=standard CpuSpecList=1                                                                                                                                           
  n1    Tags[tag1]                   default     ,                                                                                                                                                                        
  n1    Tags[arch]                   --          x86_64                                                                                                                                                                   
  n1    Tags[tag2]                   default     val2                                                                                                                                                                     
  n1    Tags[SlurmCpuSpecList]       default     1,33                                                                                                                                                                     
  n1    Tags[SlurmFeature]           default     standard                                                                                                                                                                 
  n1    Tags[SlurmSpuSpecList]       default     1,33                                                                                                                                                                     
  n1    Tags[othervar]               default     33'
```

And here's the same node shown with the PR applied:

```
[janderson@rocky main]$ sudo wwctl node list -a n1
  NODE  FIELD                        PROFILE     VALUE                                                                                                                                                                    
  n1    Id                           --          n1                                                                                                                                                                       
  n1    Comment                      SUPERSEDED  a node comment,This profile is automatically included for each node                                                                                                      
  n1    ClusterName                  --          mycluster                                                                                                                                                                
  n1    ContainerName                default     rockylinux-8                                                                                                                                                             
  n1    Ipxe                         --          (default)                                                                                                                                                                
  n1    RuntimeOverlay               --          (generic)                                                                                                                                                                
  n1    SystemOverlay                --          wwinit,fstab                                                                                                                                                             
  n1    Root                         --          (initramfs)                                                                                                                                                              
  n1    Init                         --          (/sbin/init)                                                                                                                                                             
  n1    Kernel.Args                  SUPERSEDED  crashkernel=no vga=791 net.naming-scheme=v238,crashkernel=no vga=791 net.naming-scheme=v238 systemd.unified_cgroup_hierarchy=0 systemd.legacy_systemd_cgroup_controller  
  n1    Profiles                     --          default                                                                                                                                                                  
  n1    PrimaryNetDev                --          (default)                                                                                                                                                                
  n1    NetDevs[default].Type        --          Ethernet                                                                                                                                                                 
  n1    NetDevs[default].OnBoot      --          true                                                                                                                                                                     
  n1    NetDevs[default].Device      --          eth0                                                                                                                                                                     
  n1    NetDevs[default].Hwaddr      --          de:ad:be:ef:00:00                                                                                                                                                        
  n1    NetDevs[default].Ipaddr      --          192.168.0.100                                                                                                                                                            
  n1    NetDevs[default].Netmask     --          255.255.255.0                                                                                                                                                            
  n1    NetDevs[default].Gateway     --          192.168.0.1                                                                                                                                                              
  n1    NetDevs[default].Primary     --          (true)                                                                                                                                                                   
  n1    NetDevs[default].Tags[DNS2]  SUPERSEDED  8.8.4.4,8.8.4.4                                                                                                                                                          
  n1    NetDevs[default].Tags[DNS]   --          8.8.8.8                                                                                                                                                                  
  n1    NetDevs[default].Tags[DNS1]  SUPERSEDED  8.8.8.8,8.8.8.8                                                                                                                                                          
  n1    Tags[othervar]               default     33'                                                                                                                                                                      
  n1    Tags[slurmdconf]             default     Feature=standard CpuSpecList=1                                                                                                                                           
  n1    Tags[tag1]                   default     ,                                                                                                                                                                        
  n1    Tags[tag2]                   default     val2                                                                                                                                                                     
  n1    Tags[arch]                   --          x86_64                                                                                                                                                                   
  n1    Tags[SlurmCpuSpecList]       default     1,33                                                                                                                                                                     
  n1    Tags[SlurmFeature]           default     standard                                                                                                                                                                 
  n1    Tags[SlurmSpuSpecList]       default     1,33
```

So, other than tags being emitted in a different order, the output appears to be the same.

What is this supposed to be doing?

File: internal/app/wwctl/node/list/main_test.go [hunk 1]
```diff
@@ -144,6 +144,199 @@ nodes:
...
+    profiles:
+    - default
+`},
+		{
+			name:    "node list profile with comment superseded",
+			args:    []string{"-a"},
+			wantErr: false,
+			stdout: `NODE  FIELD           PROFILE  VALUE                                         
+              n01   Id              --       n01                                                  
+              n01   Comment         SUPERSEDED  nodecomment
```
Comment (mslacken) [inline internal/app/wwctl/node/list/main_test.go:178]: @anderbubble You see here how the output should look like, for a superseeded comment, only the superseeded value must be shown, if the Entry is as string and not a slice. I think you used and unpatched version for you review.

**Comment by anderbubble** [comment]
I'll check it again. Thanks for the clarification.

**Comment by anderbubble** [comment]
Rebased to main.

**Comment by anderbubble** [comment]
Fixed changelog to update correct release.

**Review by anderbubble — APPROVED** [review]

**Pull request merged**